src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -230,6 +220,38 @@ protected Assignment CreateAssignmentObject(string identityType, Dictionary<stri
                 }   
             }
 
+            if (SecureStringParameters != null)
+            {
+                foreach (var key in SecureStringParameters.Keys)
+                {
+                    var kvp = SecureStringParameters[key] as Hashtable;
+                    string keyVaultId = null;
+                    string secretName = null;
+                    string secretVersion = null;
+
+                    foreach (var k in kvp.Keys)
+                    {
+                        var paramKey = k.ToString();
+
+                        if (string.Equals(paramKey, "keyVaultId", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
@filizt if we're parsing for specific values in this provided hashtable, why not just make them separate parameters for the cmdlet? This will provide a better experience for the user.

Reply:
That makes sense. I'll bring this up with our PM. Not updating the code in this PR, it'll probably be the next release.